var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -126,7 +136,7 @@ class Mfem(Package):
             description='Enable Cubit/Genesis reader')
     variant('conduit', default=False,
             description='Enable binary data I/O using Conduit')
-    variant('gzstream', default=True,
+    variant('zlib', default=True,
```

Comment:
@v-dobrev @tzanio When removing a variant, I think it’s worth trying to remove it to the minimum number of versions to preserve the behavior of older versions.
Suddenly a spec like `spack install mfem@3.4 ~gzstream` does not work and scripts have to be changed whereas the spec would still be valid.

Reply:
Sorry for the trouble with the script. This was intentional rename of the variant `gzstream`->`zlib` which is more consitant with other spack packages (which define `zlib` variants) and since `zlib` is the name of the spack package for the `zlib` library.

Originally, I tried to keep both variants, the old `gzstream` and the new `zlib` but that was not a good solution because there was no way to be backward compatible and consistent. For example, take the spec that you had: `mfem@3.4 ~gzstream`. With the addition of the `zlib` variant, this would become `mfem@3.4 ~gzstream+zlib` which you either forbid (with a `conflicts('~gzstream+zlib')`) or allow it and enable `ZLIB` support in mfem when at least one of the variants is enabled `+gzstream` or `+zlib`. So with `mfem@3.4 ~gzstream` you end up either with a spec failure or you end up with `ZLIB` enabled in the mfem build.

At the end, I did not see a good way to handle two variants that control one option, so I decided to go for a simple rename.

Reply:
I was thinking of keeping the `gzstream` variant up to a certain version, and moving to `zlib` afterward, to that specifying be both in the spec would not be allowed.
I’m not even certain how to do that, I figured it was possible.

However, you already dug into it, so I’m satisfied since I thought it was an oversight. Also I checked and there is no package in spack specifying the `gzstream` variant. I was being picky, certainly :).